### DISCUSSION TITLE
"forge compile" issue while trying to test inherited custom error

### DISCUSSION THREAD

**Discussion by linghuccc** [foundry-rs/foundry#7222] [Help] [answered]
Happy new year! I have encountered an issue while trying to test inherited custom error. Hoping some guru could help me out.

Ownable.sol:
```
abstract contract Ownable {
    error OwnableUnauthorizedAccount(address account);
...}
```

One.sol (this contract inherits contract Ownable):
```
import {Ownable} from "./Ownable.sol";
contract One is Ownable {...}
```

Two.sol (this contract inherits contract One):
```
import {One} from "./lib/One.sol";
contract Two is One {
    error Whatever();
    function update() external view onlyOwner {
        revert Whatever();
    }
}
```

test script (test function update() in contract Two):
```
    function test_update_fail(address user) public {
        vm.assume(user != OWNER);

        vm.expectRevert(Two.OwnableUnauthorizedAccount.selector, user);
        vm.prank(user);
        _contract.update();
    }
```

While doing "`forge compile`", "`vm.expectRevert(Two.Whatever.selector);`" works fine, but
"`vm.expectRevert(Two.OwnableUnauthorizedAccount.selector, user);`" will always trigger below error:
`Error (9582): Member "OwnableUnauthorizedAccount" not found or not visible after argument-dependent lookup in type(contract Two).`

Thank you for your time. Look forward to your help!

**Comment by JoaoMorais96** [comment]
Same problem, have you found any solution?

**Reply by linghuccc** [reply]
not yet

**Comment by pyk** [comment, marked as answer]
You can use `Ownable.OwnableUnauthorizedAccount.selector` directly

```solidity
vm.expectRevert(Ownable.OwnableUnauthorizedAccount.selector, user);
```

**Reply by JoaoMorais96** [reply]
This works thanks! But the abi.encodeWithSelector is missing and that is why it was failing in my case.
The correct code is:
```
        vm.expectRevert(
            abi.encodeWithSelector(
                Ownable.OwnableUnauthorizedAccount.selector,
                address(user)
            )
        );
```

**Reply by linghuccc** [reply]
Oh, my god! How stupid I am!! I tried that before, but DID NOT "import Ownable"!!!
So I thought that was not feasible... My holy &^%^$%&&*&*()& 
Thank you so much, man! You are the best!!
